packages/web3-shared/src/hooks/useFungibleTokenBalance.ts

```diff
@@ -1,23 +1,18 @@
 import { unreachable } from '@dimensiondev/kit'
 import { EthereumTokenType } from '../types'
 import { useERC20TokenBalance } from './useERC20TokenBalance'
-import { useERC721TokenBalance } from './useERC721TokenBalance'
 import { useNativeTokenBalance } from './useNativeTokenBalance'
 
-export function useTokenBalance(type: EthereumTokenType, address?: string) {
+export function useFungibleTokenBalance(type: EthereumTokenType.Native | EthereumTokenType.ERC20, address?: string) {
     const r1 = useNativeTokenBalance()
     const r2 = useERC20TokenBalance(type === EthereumTokenType.ERC20 ? address : undefined)
-    const r3 = useERC721TokenBalance(type === EthereumTokenType.ERC721 ? address : undefined)
+
     const type_ = type
     switch (type_) {
         case EthereumTokenType.Native:
             return r1
         case EthereumTokenType.ERC20:
             return r2
-        case EthereumTokenType.ERC721:
```

Comment:
It's wried to handle (Non)fungible tokens in one hook, and there's no such business scene so far.